### PR TITLE
 change errors.Errorf  to *.Wrapf and *.Wrap

### PR DESCRIPTION
**Pull request by developerfred**
fixes: #236

Fix
https://github.com/tellor-io/TellorMiner/issues/236#issuecomment-729972707

### CODE REVIEW

File: cmd/tellor/argTypes.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ type ETHAddress struct {
 func (a *ETHAddress) Set(v string) error {
 	valid := common.IsHexAddress(v)
 	if !valid {
-		return errors.Errorf("%s is not a valid etherum address format", v)
+		return errors.Errorf("%s valid etherum address format", v)
```
Comment (krasi-georgiev) [inline cmd/tellor/argTypes.go:44]:
```suggestion
		return errors.Errorf("invalid etherum address:%v", v)
```

It is more structured when we have the param at the end.

File: pkg/apiOracle/valueOracle.go [hunk 2]
```diff
@@ -110,14 +110,14 @@ func EnsureValueOracle() error {
...
-			return errors.Errorf("file %s stat error: %v", historyPath, err)
+			return errors.Wrapf(err, "stats for file: %v", historyPath)
 		}
 	}
 
 	if exists {
 		byteValue, err := ioutil.ReadFile(historyPath)
 		if err != nil {
-			return errors.Errorf("failed to read psr file @ %s: %v", historyPath, err)
+			return errors.Wrapf(err, "failed to read psr file @: %v", historyPath)
```
Comment (krasi-georgiev) [inline pkg/apiOracle/valueOracle.go:120]:
```suggestion
			return errors.Wrapf(err, "read psr file @: %v", historyPath)
```

File: pkg/config/config.go [hunk 3]
```diff
@@ -132,7 +132,7 @@ const PrivateKeyEnvName = "ETH_PRIVATE_KEY"
 func ParseConfig(path string) error {
 	data, err := ioutil.ReadFile(path)
 	if err != nil {
-		return errors.Errorf("failed to open config file %s: %v", path, err)
+		return errors.Wrapf(err, "to open config file: %v", path)
```
Comment (krasi-georgiev) [inline pkg/config/config.go:135]:
```suggestion
		return errors.Wrapf(err, "open config file: %v", path)
```

File: pkg/config/config.go [hunk 4]
```diff
@@ -141,7 +141,7 @@ func ParseConfig(path string) error {
 func ParseConfigBytes(data []byte) error {
 	err := json.Unmarshal(data, &config)
 	if err != nil {
-		return errors.Errorf("failed to parse json: %s", err.Error())
+		return errors.Wrap(err, "to parse json")
```
Comment (krasi-georgiev) [inline pkg/config/config.go:144]:
```suggestion
		return errors.Wrap(err, "parse json")
```

File: pkg/config/config.go [hunk 5]
```diff
@@ -169,15 +169,15 @@ func ParseConfigBytes(data []byte) error {
 
 	err = validateConfig(&config)
 	if err != nil {
-		return errors.Errorf("validation failed: %s", err)
+		return errors.Wrap(err, "validation")
```
Comment (krasi-georgiev) [inline pkg/config/config.go:172]:
```suggestion
		return errors.Wrap(err, "config validation")
```

File: pkg/config/config.go [hunk 6]
```diff
@@ -189,18 +189,18 @@ func validateConfig(cfg *Config) error {
 	} else {
 		b, err = hex.DecodeString(cfg.PrivateKey)
 		if err != nil || len(b) != 32 {
-			return errors.Errorf("expecting 64 hex character private key, got \"%s\"", cfg.PrivateKey)
+			return errors.Wrapf(err, "expecting 64 hex character private key, got \"%s\"", cfg.PrivateKey)
 		}
 		if len(cfg.ContractAddress) != 42 {
-			return errors.Errorf("expecting 40 hex character contract address, got \"%s\"", cfg.ContractAddress)
+			return errors.Wrapf(err, "expecting 40 hex character contract address, got \"%s\"", cfg.ContractAddress)
```
Comment (krasi-georgiev) [inline pkg/config/config.go:195]:
```suggestion
			return errors.Errorf("expecting 40 hex character contract address, got \"%s\"", cfg.ContractAddress)
```
There is no error to wrap here. You are printing the error from the last if block

The same in few other places

**Review by krasi-georgiev — COMMENTED** [review]
Thanks for the update, left few comments, please have a look at all other occurrences and continue in the same direction.

**Comment by developerfred** [comment]
@krasi-georgiev  thanks for review! I make changes.

**Comment by krasi-georgiev** [comment]
please change the rest of the PR according to my comments as these changes apply in many different places not just the ones I pointed.

**Comment by krasi-georgiev** [comment]
ping when ready for another review or if you get stuck making all tests pass.

File: cmd/tellor/argTypes.go [hunk 7]
```diff
@@ -63,10 +63,10 @@ func (b *EthereumInt) Set(v string) error {
 	g := new(big.Int)
 	_, ok := g.SetString(v, 10)
 	if !ok {
-		return errors.Errorf("%s is not a valid integer", v)
+		return errors.Errorf("invalid integer: %v", v)
 	}
 	if len(g.Bytes()) > 32 {
-		return errors.Errorf("%s is larger than 256 bits", v)
+		return errors.Errorf("invalid size: %v is larger than 256 bits", v)
```
Comment (krasi-georgiev) [inline cmd/tellor/argTypes.go:69]:
```suggestion
		return errors.Errorf("invalid size larger than 256 bits:%v", v)
```
it is common practise to put arguments at the end.

File: pkg/db/remoteRequest.go [hunk 8]
```diff
@@ -72,7 +72,7 @@ func createRequest(dbKeys []string, values [][]byte, signer RequestSigner) (*req
 	}
 	if sig == nil {
 		log.Error("Signature was not generated")
```
Comment (krasi-georgiev) [inline pkg/db/remoteRequest.go:74]:
```suggestion
```
not related to the PR , but we can remove this log entry, no point on having an error and a log, it is kind of duplicate.

File: pkg/db/remoteRequest.go [hunk 9]
```diff
@@ -72,7 +72,7 @@ func createRequest(dbKeys []string, values [][]byte, signer RequestSigner) (*req
...
-		return nil, errors.Errorf("Could not generate a signature for  hash: %v", hash)
+		return nil, errors.Wrapf(err, "Could not generate a signature for  hash: %v", hash)
```
Comment (krasi-georgiev) [inline pkg/db/remoteRequest.go:75]:
```suggestion
		return nil, errors.Wrapf(err, "generate a signature for  hash: %v", hash)
```
we don't need words like this, just need to describe when the error occurred. It is obvious that there is `couldn't` :smile: 

Also, all errors are lower case and no full stop at the end.

**Comment by developerfred** [comment]
@krasi-georgiev  double check in files and test pass done.

File: pkg/ops/stakeOps.go [hunk 10]
```diff
@@ -74,24 +74,24 @@ func Deposit(ctx context.Context, logger log.Logger) error {
 	copy(dat32[:], dat)
 	stakeAmt, err := tmaster.GetUintVar(nil, dat32)
 	if err != nil {
-		return errors.Errorf("fetching stake amount failed: %s", err.Error())
+		return errors.Wrap(err, "fetching stake amount")
 	}
 
 	if balance.Cmp(stakeAmt) < 0 {
-		return errors.Errorf("insufficient balance (%s), mining stake requires %s TRB",
+		return errors.Wrapf(err, "insufficient balance (%s), mining stake requires %s TRB",
```
Comment (krasi-georgiev) [inline pkg/ops/stakeOps.go:81]: Where is this error from? Please be careful about such cases as with this the error will be masked and not even returned.

**Review by krasi-georgiev — COMMENTED** [review]
We are going in a good direction, thanks.
Please review my comments and check the rest of the pr to make sure you apply the same principles everywhere.

File: pkg/ops/stakeOps.go (hunk 10 above)
Comment (developerfred) [reply pkg/ops/stakeOps.go:81]: > Where is this error from? Please be careful about such cases as with this the error will be masked and not even returned.

I will be careful. thanks.

File: scripts/opencl/opencl.go [hunk 11]
```diff
@@ -36,28 +36,28 @@ func compileSources() (string, error) {
...
-		return "", errors.Errorf("missing %s file", headerFile)
+		return "", errors.Wrapf(err, "missing %s file", headerFile)
 	}
 	out.Write(contents[headerFile])
 	delete(contents, headerFile)
 
 	mainSrc, ok := contents[kernelFile]
 	if !ok {
-		return "", errors.Errorf("missing %s file", kernelFile)
+		return "", errors.Wrapf(err, "missing %s file", kernelFile)
```
Comment (krasi-georgiev) [inline scripts/opencl/opencl.go:60]: Where is this error coming from? Please double-check the entire PR for such issues.

Comment (developerfred) [reply scripts/opencl/opencl.go:60]: When I didn't have a declared error variable I put `errors.Errorf(msg)`
I will change it to `Wrapf` it is better to have control of errors.

Comment (developerfred) [reply scripts/opencl/opencl.go]: I put missing kernel and missing header

Comment (krasi-georgiev) [reply scripts/opencl/opencl.go]: The question still stands, where does this `err` variable come from?

File: scripts/opencl/opencl.go [hunk 12]
```diff
@@ -36,28 +36,28 @@ func compileSources() (string, error) {
...
 		if err != nil {
-			return "", errors.Errorf("couldn't read file %s: %s", filepath, err.Error())
+			return "", errors.Wrapf(err, "read file: %s", filepath)
 		}
 		contents[file.Name()] = data
 	}
 	var out strings.Builder
 	if _, ok := contents[headerFile]; !ok {
-		return "", errors.Errorf("missing %s file", headerFile)
+		return "", errors.Wrapf(err, "missing header: %s file", headerFile)
```
Comment (krasi-georgiev) [inline scripts/opencl/opencl.go:53]: The question still stands. Where does this `err` variable come from?

Comment (developerfred) [reply scripts/opencl/opencl.go:53]: ah got it, when there is no variable defined `err` in the function I can use `errors.ErrorF`

another thing i changed some err.Error () to errors lib is it prudent to do this?

**Comment by krasi-georgiev** [comment]
for the error you can run `make lint` locally to find out why it happens

**Comment by krasi-georgiev** [comment]
btw in few minutes will merge another big PR that does some other global refactorings so there will be some conflict to be resolved and unfortunately we can't avoid it, but shouldn't be too difficult to resolve.

File: scripts/opencl/opencl.go [hunk 13]
```diff
@@ -36,28 +35,28 @@ func compileSources() (string, error) {
...
-		return "", errors.Errorf("missing %s file", headerFile)
+		return "", errors.Errorf("missing header: %s file", headerFile)
 	}
 	out.Write(contents[headerFile])
 	delete(contents, headerFile)
 
 	mainSrc, ok := contents[kernelFile]
 	if !ok {
-		return "", errors.Errorf("missing %s file", kernelFile)
+		return "", errors.Errorf("missing kearnal: %s file", kernelFile)
```
Comment (krasi-georgiev) [inline scripts/opencl/opencl.go:59]:
```suggestion
		return "", errors.Errorf("missing kernel file:%s", kernelFile)
```
please put arguments at the end to make these easier to read, the same in few other places.

File: pkg/util/logConfig.go [hunk 14]
```diff
@@ -32,10 +32,10 @@ func ParseLoggingConfig(file string) error {
 	if len(file) > 0 {
 		info, err := os.Stat(file)
 		if os.IsNotExist(err) {
-			return errors.Errorf("loggingConfigPath references an invalid file at: %s", file)
+			return errors.Wrapf(err, "loggingConfigPath references an invalid file at: %s", file)
 		}
 		if info.IsDir() {
-			return errors.Errorf("logging config file %s is a directory", file)
+			return errors.Wrapf(err, "logging config file %s is a directory", file)
```
Comment (krasi-georgiev) [inline pkg/util/logConfig.go:38]: Again this error is set somewhere else so should be used here. Please check the entire PR for these again.

File: pkg/tracker/currentVariables.go [hunk 15]
```diff
@@ -90,7 +90,7 @@ func (b *CurrentVariablesTracker) Exec(ctx context.Context) error {
 
 	err = DB.Put(db.TotalTipKey, []byte(hexutil.EncodeBig(returnNewVariables.Tip)))
 	if err != nil {
-		return errors.Wrap(err, "totaltip put")
+		return errors.Wrap(err, "taltip put")
```
Comment (krasi-georgiev) [inline pkg/tracker/currentVariables.go:93]:
```suggestion
		return errors.Wrap(err, "total tip put")
```

File: pkg/pow/solutionHandler.go [hunk 16]
```diff
@@ -89,7 +89,7 @@ func (s *SolutionHandler) Submit(ctx context.Context, result *Result) (*types.Tr
 					}
 					continue
 				}
-				return nil, errors.Errorf("no value in database,  reg id: %v", challenge.RequestIDs[i].Uint64())
+				return nil, errors.Wrapf(err, "no value in database,  reg id:%v", challenge.RequestIDs[i].Uint64())
```
Comment (krasi-georgiev) [inline pkg/pow/solutionHandler.go:92]: Where is this error from?
```suggestion
				return nil, errors.Errorf("no value in database,  reg id:%v", challenge.RequestIDs[i].Uint64())
```

**Review by krasi-georgiev — APPROVED** [review]

**Comment by krasi-georgiev** [comment]
Thanks!

**Pull request merged**